server/src/main/java/org/elasticsearch/env/BuildVersion.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.env;
+
+import org.elasticsearch.Build;
+import org.elasticsearch.Version;
+import org.elasticsearch.internal.BuildExtension;
+import org.elasticsearch.plugins.ExtensionLoader;
+
+import java.util.ServiceLoader;
+
+/**
+ * A pluggable version class with a minimal comparison API.
```

Comment:
This makes it sound like a generic version class, but it's not. Could we use something like "A version representing the code of Elasticsearch" or something like that? The bits about it being minimal and pluggable can be in the description below.